apple/testing/default_runner/ios_test_runner.template.sh

```diff
@@ -146,10 +158,62 @@ else
   )
 fi
 
+exit_status=0
 cmd=("%(testrunner_binary)s"
   "${runner_flags[@]}"
   "${target_flags[@]}"
   "${custom_xctestrunner_args[@]}")
-"${cmd[@]}" 2>&1
-status=$?
-exit ${status}
+"${cmd[@]}" 2>&1 || exit_status=$?
+
+if [[ "$exit_status" -ne 0 ]]; then
+  exit "$exit_status"
+fi
+
+if [[ "${COVERAGE:-}" -ne 1 ]]; then
+  # Normal tests run without coverage
+  exit 0
+fi
+
+readonly profdata="$TMP_DIR/coverage.profdata"
+xcrun llvm-profdata merge "$profraw" --output "$profdata"
+
+# llvm-cov export doesn't print any warnings/errors when files that it's trying
+# to reference don't exist. Unfortunately these are absolute paths that won't
+# be valid across machines. When using llvm-cov show it prints a warning for
+# this so we can validate we don't hit this case.
+error_file=$(mktemp)
+llvm_cov_status=0
+xcrun llvm-cov \
+  show \
+  -instr-profile "$profdata" \
+  -path-equivalence="$ROOT","$PWD" \
+  "$test_binary" \
+  @"$COVERAGE_MANIFEST" \
+  2> "$error_file" \
+  > /dev/null \
+  || llvm_cov_status=$?
+
+if [[ -s "$error_file" || "$llvm_cov_status" -ne 0 ]]; then
+  echo "error: while showing coverage report" >&2
+  cat "$error_file" >&2
+  exit 1
+fi
+
+xcrun llvm-cov \
+  export \
+  -format lcov \
+  -instr-profile "$profdata" \
+  -path-equivalence="$ROOT","$PWD" \
+  "$test_binary" \
+  @"$COVERAGE_MANIFEST" \
+  > "$COVERAGE_OUTPUT_FILE" \
```

Comment:
Do you have docs on these variables?

Reply:
don't think there is any, some are in here https://github.com/bazelbuild/bazel/blob/master/tools/test/collect_coverage.sh but I just ran `env; exit 1` in this script to see what was available. it does seem to treat these special (we just throw things in the undeclared outputs)

Reply:
TIL just go into Bazel stuff and do `env; exit 1`